sisl/io/xyz.py

```diff
@@ -98,8 +99,11 @@ def read_geometry(self, atoms=None, sc=None):
         sc : SuperCell, optional
             the supercell to be associated with the geometry
         """
+        line = self.readline()
+        if line == '':
+            return None
         # Read number of atoms
-        na = int(self.readline())
+        na = int(line)
```

Comment:
wow, I am surprised this wasn't caught in the tests... this looks like a severe bug...

Reply:
There is also another issue. What if some blank lines are present before or after a xyz-geometry block? Should the code not be able to handle such situations?

Reply:
Preferentially yes. This could easily be solved with
```python
def sile_read_multiple(..., pre_call=None, post_call=None):
    ...
   pre_call()
   r = func(...)
   post_call()
```
or similar. I think this could cover all use cases.

Reply:
Then the user would need to specify `pre_call=self.readline()` if there is a blank line before the atom number?

Could `read_geometry()` not simply read blank lines until it finds an integer?

Reply:
> Then the user would need to specify `pre_call=self.readline()` if there is a blank line before the atom number?
> 
> Could `read_geometry()` not simply read blank lines until it finds an integer?

but for more complex files which contains `xyz` you wouldn't know if that integer starts the geometry, or some other segment. It is a rabbit hole ;)

I think the above is the most stable one, after all it is not something the user sees, it is the file implementor who knows how its handled.